packages/react/src/hooks/contracts/useContractReads.ts

```diff
@@ -20,29 +20,44 @@ export type UseContractReadsConfig = QueryConfig<ReadContractsResult, Error> &
     watch?: boolean
   }
 
-export const queryKey = ([{ contracts, overrides }, { blockNumber }]: [
-  ReadContractsConfig,
-  { blockNumber?: number },
-]) =>
+export const queryKey = ([
+  { allowFailure, contracts, overrides },
+  { blockNumber, chainId },
+]: [ReadContractsConfig, { blockNumber?: number; chainId?: number }]) =>
   [
     {
       entity: 'readContracts',
-      contracts,
+      allowFailure,
       blockNumber,
+      chainId,
+      contracts,
       overrides,
     },
   ] as const
 
+const queryKeyHashFn = ([queryKey_]: ReturnType<typeof queryKey>) => {
+  const { contracts, ...rest } = queryKey_
+  const contracts_ = contracts?.map((contract) => {
+    // Exclude the contract interface from the serialized query key.
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { contractInterface, ...rest } = contract
```

Comment:
`contractInterface` bloated the hashed query key and caused a weird bug where data would persist on the wrong query key.